### ISSUE TITLE
[a11y]: Tearsheet with navigation

### ISSUE THREAD

**Issue by allisonishida** [carbon-design-system/ibm-products#4989]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

^2.31.0

### React version

^18.2.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Version 3.1.70

### Description

Using the `Tearsheet` component with `Tabs`, etc. in navigation field poses accessibility violations. 

I have the `Tabs`, `TabList`, and `Tab` in the `navigation` section, but the `Tabs`, `TabPanels`, and `TabPanel` in the content of the Tearsheet. This results in multiple violations
```
The 'id' "ccs-:r49:-tabpanel-0" specified for the ARIA property 'aria-controls' value is not valid
```
and
```
The 'id' "ccs-:rtp:-tab-0" specified for the ARIA property 'aria-labelledby' value is not valid
```
because I don’t have the `TabPanels` under the same parent `Tabs` component within `navigation`.

### Reproduction/example

https://codesandbox.io/p/devbox/shy-sun-kl9t29?file=%2Fsrc%2FExample%2FExample.jsx%3A64%2C13-72%2C24

### Steps to reproduce

Scan the page using the IBM Equal Access Accessibility Checker

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
```[tasklist]
### Tasks
- [ ] Audit tab usage (`navigation` prop) within Tearsheet
- [ ] Propose and/or implement any changes
```

**Comment by davidmenendez** [comment]
@allisonishida thanks for bringing this up. we've confirmed that currently there's no fix for this particular implementation. @ariellalgilmore has outlined the issue in more depth [here](https://github.com/carbon-design-system/carbon/issues/10313#issuecomment-2155168858). we're now evaluating a few different solutions to solve this.

**Comment by davidmenendez** [comment]
@allisonishida @elycheea i've got a PR open now to address this #5971 

`Tearsheet` is a great candidate for an example of a component riddled with recipe code. this particular nav implementation version being one of them. definitely something we should look at as i think the stories could be cleaned up a tad.

Allison, this fix should be merged shortly 👍
